### DISCUSSION TITLE
Version 3.5.1 is not found when checking for updates from version 3.4.3

### DISCUSSION THREAD

**Discussion by adamralph** [gitextensions/gitextensions#9339] [Q&A] [answered]
## Current behaviour

I'm using version 3.4.3.9999. When I click Help -> Check for updates, no updates are found.

## Expected behaviour

Version 3.5.1 should be found — https://github.com/gitextensions/gitextensions/releases/tag/v3.5.1

## Steps to reproduce

- Open version 3.4.3.9999
- Help -> Check for updates

## Screenshots

![image](https://user-images.githubusercontent.com/677704/124387484-2f0b0080-dcdf-11eb-9490-342782ea4d9f.png)

![image](https://user-images.githubusercontent.com/677704/124387504-40eca380-dcdf-11eb-8466-fecb739bcc4b.png)

## Did this work in previous version of GitExtensions

Not sure.

## Environment

- GitExtensions version:
- GIT version: `git version 2.32.0.windows.1`
- OS version: `10.0.19042 Build 19042`
- .NET version: 4.8.4360.0

**Comment by pmiossec** [comment, marked as answer]
That's not a bug but a team choice to decide when the next update is proposed to the update check.

v3.5 has still not been proposed because we had concerning crash reports. They should have been fixed with v3.5.1 so if returns are good maybe sometime it will be proposed.

Closing because working as desired.

**Comment by adamralph** [comment]
OK, but can you see why I was led to raise this as a bug? If the UI told me what you just told me, then I wouldn't have raised it.

**Comment by RussKie** [comment]
The UI told you that- no update available. Once the team is satisfied with
the quality, the auto update will be available.

**Comment by vbjay** [comment]
https://github.com/gitextensions/gitextensions/tree/configdata.

**Comment by adamralph** [comment]
- The UI tells me there is no update available past 3.4.3.
- GitHub tells me that 3.5.1 is available.

Can you see why this looks like a bug to a user?

**Comment by vbjay** [comment]
No. A lot of places do a release that lets you test and when the bugs are
delt with they let the auto update functionality kick in. That is how it is
done here. You can try the latest release by manually downloading.

**Comment by pmiossec** [comment]
@adamralph i understand that it could be strange from an outside point of view but what you are asking, to better handle this case by more information or something else, requires a lot of effort for very few added value...

**Reply by adamralph** [reply]
@vbjay @pmiossec this looks a bit weird after being moved from an issue to a discussion since there is only really on thread of conversation, but to continue —

> No. A lot of places do a release that lets you test and when the bugs are
delt with they let the auto update functionality kick in. That is how it is
done here. You can try the latest release by manually downloading.

> i understand that it could be strange from an outside point of view but what you are asking, to better handle this case by more information or something else, requires a lot of effort for very few added value...

It does indeed look a bit strange to being doing a "partial" release (to GitHub releases), and then only a "full" release (to the auto-update client, etc.) after validating that release. For example, what if I'm installing GHE on a new machine? I might go to the GitHub releases page, install 3.5.1, blissfully unware that it has not been sanctioned as "good enough" to be enabled for auto-update.

However, I don't want to criticise the current approach with offering an alternative:

Why not use release candidates? For example:

- Release 3.5.0-rc.1 (to GitHub releases and any other pre-RTM channels)
- Do all the various testing, etc. (let's imagine you find a bug)
- Release 3.5.0-rc.2 (to GitHub releases and any other pre-RTM channels)
- Do all the various testing, etc. (let's imagine you find no bugs)
- Release 3.5.0 to _all_ channels

This would avoid users seeing 3.5.0 available in one place (e.g. GitHub releases) but not available in others (e.g. auto-update). The RTM is either released or not released. It would also simplify the auto-update — no need for any special logic to handle whether it should or should not be available. The auto-update can just grab whatever RTM is available. It would also avoid having to rev the patch number (from 3.5.0 to 3.5.1) while the package is in the "partial" (GitHub releases only) state while you iron out bugs. The auto-update would only see only clean RTM releases, rather than a jump, from say, 3.4.3 to 3.5.1, even though 3.5.0 was never "fully" released for users (at least those using auto-update) for the bug that was fixed in 3.5.1 to observable by them.

**Comment by gerhardol** [comment]
Release candidates are used and the normal procedure is to promote releases soon after the first release.
For 3.5 RC, there were hardly any feedback at all though. After 3.5.0 was released there were a bunch of feedback issues. 
There was especially #9166. It took some time to analyze that issue. The issue turned out to not really be a new issue in 3.5 but due to the user used other features and to new error reporting in 3.5.

We all hope this was an exception and that the 3.5 scenario is not repeated.
The limited time that is available is therefore in maintainers point of view better spent elsewhere.

But the project is open to contributions.
